packages/crayons-core/src/components/icon/readme.md

```diff
@@ -23,28 +30,184 @@ import ReactDOM from "react-dom";
+fwIconRegisterLibrary('feather', {
+        resolver: () => `https://cdn.jsdelivr.net/npm/feather-icons@4.28.0/dist/icons`,
+        mutator: (svg,'all') => svg.setAttribute('fill', 'currentColor')
+});
+</script>
+</head>
+<body>
+    <fw-icon name="alert" color="red" size="30" ></fw-icon>
+    <fw-icon name="feather" library="feather" color="red" size="30" ></fw-icon>
+</body>
+</html>
+```
+</code-block>
+
+<code-block title="React-FwIconLibrary">
+``` jsx
+import React from 'react';
+import './App.css';
+import { add_contact } from '@freshworks/crayons-icon';
+import { FwIcon, fwIconRegisterLibrary } from '@freshworks/crayons/react';
+
+fwIconRegisterLibrary('feather', {
+        resolver: () => `https://cdn.jsdelivr.net/npm/feather-icons@4.28.0/dist/icons`,
+        mutator: (svg,name) => (name==='feather') ? svg.setAttribute('fill', 'currentColor') : false
+        
+});
+fwIconRegisterLibrary('heroicons', {
+        resolver: () => `https://cdn.jsdelivr.net/npm/heroicons@0.4.2/outline`,
+        mutator: (svg,name) => svg.setAttribute('fill', 'currentColor')
+});
+fwIconRegisterLibrary('custom', {
+        resolver: () => `assets/icons`
+});
+
+function App() {
+
+    return ( 
+        <div >
+                <div >
+                <FwIcon name = "alert"
+                        color = 'red'
+                        library = "custom" 
+                />
+                <FwIcon name = "feather"
+                        library="feather"
+                        color = 'red' 
+                />
+                <FwIcon name = "alert"
+                        color = 'blue'
+                        width = "20"
+                        height = "20" 
+                />
+                <FwIcon dataSvg = { add_contact }
+                        color = 'blue'
+                        size = "30" 
+                />
+                <FwIcon name = "feather"
+                        src = "https://cdn.jsdelivr.net/npm/feather-icons@4.28.0/dist/icons"
+                        color = 'blue'
+                        size = "30" 
+                />
+                </div> 
+        </div>
+    );
+}
+export default App;
+```
+</code-block>
+</code-group>
+
+
+
+## Importing Icons from '@freshworks/crayons-icon' as Inline-SVGs. Supports Tree-Shaking.
+
+We may also import Crayons Icons from '@freshworks/crayons-icon'. These are in form of JS Exports. This helps you to use inline SVG with Tree-Shakeable Imports
+This way, you may choose to alter the SVG data and push it for re-render as per your project needs.
+
+<code-group>
+<code-block title="HTML">
+```html 
+<fw-icon data-svg={ header } size="20" ></fw-icon>
+```
+</code-block>
+
+<code-block title="React">
+```jsx
+import React from "react";
+import ReactDOM from "react-dom";
+import { FwIcon } from "@freshworks/crayons/react";
+import { header } from '@freshworks/crayons-icon';
+
+function App() {
+  return (<div>
+  <FwIcon data-svg={ header } size="20" ></FwIcon>
```

Comment:
This would break in a typescript app . Please pass the correct type as value for the props.

Reply:
Fixed